### PR TITLE
[no ticket][risk=no] Create Puppeteer test results JSON file with jest-stare

### PR DESCRIPTION
**Pull request by aweng98**
Test Results Monitoring:
Step 1: Upload Puppeteer test results JSON files from CircleCI to Google storage bucket.

Approach: Use `jest-stare` to automate saving of test results in JSON format. Then use `gsutil` CLI to upload JSON files from CircleCI to storage bucket.  

storage bucket: [gs://circleci-puppeteer-test-results](https://console.cloud.google.com/storage/browser/circleci-puppeteer-test-results;tab=objects?forceOnBucketsSortingFiltering=false&authuser=0&project=all-of-us-workbench-test&forceOnObjectsSortingFiltering=false)
project: all-of-us-workbench-test

Example:
https://app.circleci.com/pipelines/github/all-of-us/workbench/5438/workflows/e86e0895-2178-4c75-ad6a-34a5b9f8ff0c/jobs/107658/artifacts

Two JSON files to be uploaded to the storage bucket. `circleci-build-107658-0-test-results.json` and `circleci-build-107658-0-test-results.json`

Screen Shot 2020-11-02 at 1 56 39 PM

### CODE REVIEW

File: e2e/tests/cohorts/cohorts-create.spec.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ import {findOrCreateWorkspace, signIn} from 'utils/test-utils';
 import {waitForText, waitWhileLoading} from 'utils/waits-utils';
 
 
-describe('User can create new Cohorts', () => {
+describe.skip('User can create new Cohorts', () => {
```
Comment (aweng98) [inline e2e/tests/cohorts/cohorts-create.spec.ts:14]: skipped most tests for testing quicker. Changes will be reverted back before merge.

File: .circleci/config.yml [hunk 2]
```diff
@@ -301,6 +301,16 @@ commands:
...
             yarn test:ci $(cat /tmp/tests-to-run)
           no_output_timeout: 15m
+      - run:
+          name: Create circleci_parameters.json
+          working_directory: ~/workbench/e2e
+          command: |
+            echo "{ \"circleBranch\": \"${CIRCLE_BRANCH}\", \"circleJob\": \"${CIRCLE_JOB}\", \"circleBuild\": \"${CIRCLE_BUILD_NUM}\", \"circleBuildUrl\": \"${CIRCLE_BUILD_URL}\", \"circleNodeIndex\": \"${CIRCLE_NODE_INDEX}\" }" \
+              > ./logs/circleci_parameters.json
+      - run:
+          name: Combine circleci_parameters.json and test-results.json
```
Comment (calbach) [inline .circleci/config.yml:311]: nit: I would not re-encode the file names here in the description, e.g. "Add CircleCI job data to Junit test results".

File: .circleci/config.yml [hunk 3]
```diff
@@ -301,6 +301,16 @@ commands:
             circleci tests glob "tests/**/*.spec.ts" | circleci tests split --split-by=timings > /tmp/tests-to-run
             yarn test:ci $(cat /tmp/tests-to-run)
           no_output_timeout: 15m
+      - run:
+          name: Create circleci_parameters.json
+          working_directory: ~/workbench/e2e
+          command: |
+            echo "{ \"circleBranch\": \"${CIRCLE_BRANCH}\", \"circleJob\": \"${CIRCLE_JOB}\", \"circleBuild\": \"${CIRCLE_BUILD_NUM}\", \"circleBuildUrl\": \"${CIRCLE_BUILD_URL}\", \"circleNodeIndex\": \"${CIRCLE_NODE_INDEX}\" }" \
```
[the remark this answers is not in the document]
Comment (aweng98) [reply .circleci/config.yml:308]: Done

File: .circleci/config.yml (hunk 2 above)
Comment (aweng98) [reply .circleci/config.yml:311]: Done

**Review by aweng98 — COMMENTED** [review]
Good to know about `fdescribe`.

File: .circleci/config.yml [hunk 4]
```diff
@@ -300,6 +301,33 @@ commands:
...
+              }" \
+              > ./logs/circleci_parameters.json
+          when: always
+      - run:
+          name: Add CircleCI job data to test results json before upload json to storage bucket
+          working_directory: ~/workbench/e2e/logs
+          command: |
+            NEW_JSON=circleci-build-${CIRCLE_BUILD_NUM}-${CIRCLE_NODE_INDEX}-test-results.json
+            jq -s '.[0] * .[1]' test-results.json circleci_parameters.json > ${NEW_JSON}
+            # Remove all whitespaces
```
Comment (calbach) [inline .circleci/config.yml:323]: Why? If you really need to do this, can you just pass the "-c" flag to fq instead?

File: .circleci/config.yml [hunk 5]
```diff
@@ -300,6 +301,33 @@ commands:
...
+            cat ${NEW_JSON} | tr '\n' ' ' > tmp.json && mv tmp.json ${NEW_JSON}
+            # Upload json to storage bucket
+            PROJECT_NAME=all-of-us-workbench-test
+            STORAGE_BUCKET=gs://circleci-puppeteer-test-results
+            gcloud config set project ${PROJECT_NAME}
```
Comment (calbach) [inline .circleci/config.yml:328]: This has implicit side-effects on other gcloud commands, in general I recommend avoiding setting this in most cases. This line also should not be necessary at all for running the following gsutil command. I would remove it and the project name.

If you still have problems after this, please let me know, but you can instead pass an explicit project ID flag to gsutil.

**Review by calbach — APPROVED** [review]
Looks good, some minor comments

File: .circleci/config.yml (hunk 5 above)
Comment (aweng98) [reply .circleci/config.yml]: 👍 
Addressing latest feedback: Use jq `-c`, and `gsuite cp` works after removing set project. Thank you CH.

**Pull request merged**